### DISCUSSION TITLE
How can I compile my winUI 3 Blanked application on uno?

### DISCUSSION THREAD

**Discussion by samsunlu55atakum** [unoplatform/uno#13608] [Q&A]
How can I compile my winUI 3 Blanked application on uno? Does uno support this ?

**Comment by jeromelaban** [comment]
Thanks for the question. Can you explain a bit more what you need to do?

**Reply by samsunlu55atakum** [reply]
I got a winUI 3 application (blanked app). I am willing to compile the app on uno platform. I am expecting that I won't need to rewrite the winUI 3 code or copy paste page xaml and code behind. I just want to learn is that possible to compiling winUI 3 application on uno to migrate cross-platform. Because i want to my winUI 3 application(not library or not just xaml) run on linux(skia.gtk.framebuffer). I tried to paste winUI 3's .csproj file into uno platform app. When I right click on winUI 3 app and then "set as startup project", it is working on windows. But how can I run it on mac or linux without rewrite or copy paste method?  Is that possible?

**Reply by jeromelaban** [reply]
Thanks for the details. The best way to do this is to create a new project using the Uno Platform app template, and move your code to the shared library. This way, you'll get your code to be available on all targets, WinUI and Framebuffer included.

**Reply by samsunlu55atakum** [reply]
Which files do I have to move from winUI 3 to uno platform app inside .Shared file?

**Reply by jeromelaban** [reply]
All your relevant files, such as the `.xaml` and `.cs` files that you added to your original project.

**Reply by samsunlu55atakum** [reply]
@jeromelaban including properties, dependencies?

**Reply by samsunlu55atakum** [reply]
@jeromelaban and also winUI 3 is using page instead window There are issues

**Reply by jeromelaban** [reply]
Indeed, you need to pull anything needed to build your app. The `Window` dependency is a bit different in Uno, but you can find the structure when creating a blank app in the Uno Platform wizard.

**Reply by samsunlu55atakum** [reply]
then, there is no a way compiling winUI 3 application automatically on uno platform for linux? I mean opening a winUI 3 app inside uno platform and compile it to linux? Like there are 3 heads such as uwp, wasm and linux. I don't want to change any code inside the winUI 3 app

**Reply by jeromelaban** [reply]
Thanks. It seems that you created an app using the templates for UWP not WinUI. Make sure to follow the [getting started documentation here](https://platform.uno/docs/articles/get-started-vs-2022.html), which shows how to create a WinUI-based app.

You will only need to move your code to the WinUI uno structure, and the WinUI app will continue to work as is.

**Reply by samsunlu55atakum** [reply]
@jeromelaban  the left one is winUI 3, the right one is uno platform app. Which head Should I move my winUI 3 project? Base or z_UnoApp_1 ? which one?
![Ekran Görüntüsü (29)](https://github.com/unoplatform/uno/assets/143163037/8f8274d4-ab3f-4482-898a-34c0056c4b2a)

**Reply by samsunlu55atakum** [reply]
@jeromelaban  and can I reference my winUI 3 project instead z_UnoApp_1 ? Will it work?

**Reply by samsunlu55atakum** [reply]
"When you create a new Uno Platform application, you can choose to create a WinUI 3-compatible application (using the WinUI 3 API, and building with WinUI 3 on the Windows head project) instead of a UWP XAML-compatible application [using the dotnet new templates](https://platform.uno/docs/articles/get-started-dotnet-new.html#uno-platform-blank-application-for-winappsdk---winui-3)." in the link https://platform.uno/docs/articles/uwp-vs-winui3.html says that you can choose to create a winUI 3 app inside uno and (i guess) compiling. But there is no way how to do that.

**Reply by samsunlu55atakum** [reply]
and this link : https://l24.im/8PB covers winUI 3 ?? because it covers uwp and winUI.

**Reply by samsunlu55atakum** [reply]
i can't see enable UI debugging for winUI 3 project in preview section.

**Reply by jeromelaban** [reply]
> i can't see enable UI debugging for winUI 3 project in preview section.

I'm sorry, I don't understand what you mean. What are you referring to?

**Reply by samsunlu55atakum** [reply]
@jeromelaban The link states that you can open WinUI 3 applications in Uno platform instead of UWP, but it doesn't provide detailed instructions on how to do it.  https://platform.uno/docs/articles/uwp-vs-winui3.html

**Reply by jeromelaban** [reply]
What do you mean by "open WinUI 3 applications" ? Are you talking about the templates to create app from scratch?

**Reply by samsunlu55atakum** [reply]
@jeromelaban 
![Ekran Görüntüsü (33)](https://github.com/unoplatform/uno/assets/143163037/4ee3f32c-9bba-4ed8-9b77-668884431149)

**Reply by jeromelaban** [reply]
Thanks. This document states that you can choose to create an app using either UWP or WinUI APIs. Choose whichever is applicable to your scenario, then import your original app code in it.

**Reply by samsunlu55atakum** [reply]
what do you mean by saying uwp or winUI api? Uwp is already winUI 2 ? Isn't it

**Reply by jeromelaban** [reply]
You can thank the Microsoft naming for that, but no. WinUI 2 is UWP compatible, and WinUI 3 is WinAppSDK compatible. When we're referring to WinUI, we're generally talking about WinUI 3 and WinAppSDK.

`Uno.UI.*` are UWP source compatible, and `Uno.WinUI.*` packages are WinUI+WinAppSDK source compatible.

**Reply by samsunlu55atakum** [reply]
I have asked before, but one day might Uno will able to support xaml island? I used xaml island to use winUI 2 custom controls inside wpf. I'm trying to migrate my wpf application to uno. I used xaml island to separate wpf from the windows system library (I'm talking about UI). Then I thought that uno would support wpf with xaml island. I need to port my wpf application to uno step by step. @jeromelaban

**Reply by jeromelaban** [reply]
I'll answer again, Uno only supports Uno Islands (not XAML islands) on the WPF head to include Uno/WinUI controls inside of a WPF app. We do not have plans to support other scenarios at this time.

There's a sample of Uno Islands here: https://github.com/unoplatform/Uno.Samples/tree/master/UI/UnoIslandsSampleApp.
